### PR TITLE
Add user registration form

### PR DESCRIPTION
**Pull request by lbodor**

### CODE REVIEW

File: src/client/app/shared/global/user-auth.service.spec.ts [hunk 1]
```diff
@@ -1,39 +0,0 @@
-import { UserAuthService } from './user-auth.service';
```
Comment (oehm-smith) [inline src/client/app/shared/global/user-auth.service.spec.ts]: We should fill this out.

File: src/client/app/user-registration/user-registration.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                            'Lazar',
+                            'Bodor',
+                            'Geoscience Australia',
+                            'Software developer',
+                            'lazar.bodor@ga.gov.au',
+                            '0451061798',
+                            'ADE1',
+                        );
+
+                        const setValue = function(value: string, formControlName: string): void {
```
Comment (oehm-smith) [inline src/client/app/user-registration/user-registration.component.spec.ts]: You might prefer the form:

    const setValue = (value:string, formControlName: string): void => { ... }

File: src/client/app/user-registration/user-registration.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import { RouterTestingModule } from '@angular/router/testing';
+import { BaseRequestOptions, ConnectionBackend, Http, HttpModule } from '@angular/http';
+import { MockBackend } from '@angular/http/testing';
+import { UserRegistrationComponent } from './user-registration.component';
+import { UserRegistrationModule } from './user-registration.module';
+import { UserAuthService, UserRegistration } from '../shared/global/user-auth.service';
+import { DialogService } from  '../shared/global/dialog.service';
+import { Observable } from 'rxjs/Observable';
+import { Subscriber } from 'rxjs';
+import * as _ from 'lodash';
```
Comment (oehm-smith) [inline src/client/app/user-registration/user-registration.component.spec.ts]: Didn't you want:

    import * as lodash from 'lodash';

Comment (lbodor) [inline src/client/app/user-registration/user-registration.component.spec.ts]: I didn't now that underscore was the adopted convention for lodash. Now I do.

File: src/client/app/user-registration/user-registration.component.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import { UserAuthService } from '../shared/global/user-auth.service';
+
+@Component({
+    moduleId: module.id,
+    selector: 'sd-user-registration',
+    templateUrl: 'user-registration.component.html',
+})
+export class UserRegistrationComponent {
+
+    userRegistrationForm = new FormGroup({
```
Comment (oehm-smith) [inline src/client/app/user-registration/user-registration.component.ts]: Why not use the builder:



    uRF = formBuilder.group({...});

Like I thought we decided on using?

File: src/client/app/shared/global/user-auth.service.ts [hunk 5]
```diff
@@ -1,7 +1,21 @@
...
+    constructor(
+        readonly firstName: string,
+        readonly lastName: string,
+        readonly organisation: string,
+        readonly position: string,
+        readonly email : string,
+        readonly phone : string,
+        readonly remarks: string,
+      ) {}
+}
```
Comment (frankfu03) [inline src/client/app/shared/global/user-auth.service.ts]: Should user provide country/state (or even address) as well?

Comment (lbodor) [inline src/client/app/shared/global/user-auth.service.ts]: Probably, but we can add those later. This is a start.

**Pull request merged**